dronegen/mac.go

```diff
@@ -0,0 +1,199 @@
+				"GITHUB_PRIVATE_KEY": {fromSecret: "GITHUB_PRIVATE_KEY"},
+			},
+			Commands: pushCheckoutCommandsDarwin(),
+		},
+		{
+			Name: "Build Mac artifacts",
+			Environment: map[string]value{
+				"GOPATH":        {raw: path.Join(p.Workspace.Path, "/go")},
+				"GOCACHE":       {raw: path.Join(p.Workspace.Path, "/go/cache")},
+				"OS":            {raw: "darwin"},
+				"ARCH":          {raw: "amd64"},
+				"WORKSPACE_DIR": {raw: p.Workspace.Path},
+			},
+			Commands: tagBuildCommandsDarwin(),
+		},
+		cleanUpExecStorageStep(p.Workspace.Path),
+		{
+			Name:        "Send Slack notification (exec)",
+			Environment: map[string]value{"SLACK_WEBHOOK_DEV_TELEPORT": {fromSecret: "SLACK_WEBHOOK_DEV_TELEPORT"}},
+			Commands: []string{
+				`
+export DRONE_BUILD_LINK="${DRONE_SYSTEM_PROTO}://${DRONE_SYSTEM_HOSTNAME}/${DRONE_REPO_OWNER}/${DRONE_REPO_NAME}/${DRONE_BUILD_NUMBER}"
+export GOOS=$(go env GOOS)
+export GOARCH=$(go env GOARCH)
+`,
+				fmt.Sprintf(`
+curl -sL -X POST -H 'Content-type: application/json' --data "{\"text\":\"Warning: %s artifact build failed for [%s] - please investigate immediately!\nBranch: %s\nCommit: %s\nLink: $DRONE_BUILD_LINK\"}" $SLACK_WEBHOOK_DEV_TELEPORT`,
+					escapedPreformatted("$GOOS-$GOARCH"),
+					escapedPreformatted("${DRONE_REPO_NAME}"),
+					escapedPreformatted("${DRONE_BRANCH}"),
+					escapedPreformatted("${DRONE_COMMIT_SHA}")),
+			},
+			When: &condition{Status: []string{"failure"}},
+		},
+	}
+	return p
+}
+
+func tagDarwinPipeline() pipeline {
+	p := newDarwinPipeline("build-darwin-amd64")
+	p.Trigger = triggerTag
+	p.Steps = []step{
+		setUpExecStorageStep(p.Workspace.Path),
+		{
+			Name: "Check out code",
+			Environment: map[string]value{
+				"WORKSPACE_DIR":      {raw: p.Workspace.Path},
+				"GITHUB_PRIVATE_KEY": {fromSecret: "GITHUB_PRIVATE_KEY"},
+			},
+			Commands: tagCheckoutCommandsDarwin(),
+		},
+		{
+			Name: "Build Mac release artifacts",
+			Environment: map[string]value{
+				"GOPATH":        {raw: path.Join(p.Workspace.Path, "/go")},
+				"GOCACHE":       {raw: path.Join(p.Workspace.Path, "/go/cache")},
+				"OS":            {raw: "darwin"},
+				"ARCH":          {raw: "amd64"},
+				"WORKSPACE_DIR": {raw: p.Workspace.Path},
+			},
+			Commands: tagBuildCommandsDarwin(),
+		},
+		{
+			Name: "Copy Mac artifacts",
+			Environment: map[string]value{
+				"WORKSPACE_DIR": {raw: p.Workspace.Path},
+			},
+			Commands: tagCopyPackageArtifactCommandsDarwin(),
+		},
+		{
+			Name: "Upload to S3",
+			Environment: map[string]value{
+				"AWS_S3_BUCKET":         {fromSecret: "AWS_S3_BUCKET"},
+				"AWS_ACCESS_KEY_ID":     {fromSecret: "AWS_ACCESS_KEY_ID"},
+				"AWS_SECRET_ACCESS_KEY": {fromSecret: "AWS_SECRET_ACCESS_KEY"},
+				"AWS_REGION":            {raw: "us-west-2"},
+				"WORKSPACE_DIR":         {raw: p.Workspace.Path},
+			},
+			Commands: uploadToS3CommandsDarwin(),
+		},
+		cleanUpExecStorageStep(p.Workspace.Path),
+	}
+	return p
+}
+
+func pushCheckoutCommandsDarwin() []string {
```

Comment:
nit: let's keep naming consistent and name all the helper functions `func darwin*()` instead of `func *Darwin()`